File: auth/src/main/java/com/harrish/auth/dto/AuthenticationResponse.java
package com.harrish.auth.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public record AuthenticationResponse(
        @JsonProperty("access_token")
        String accessToken,

        @JsonProperty("refresh_token")
        String refreshToken,

        @JsonProperty("token_type")
        String tokenType,

        @JsonProperty("expires_in")
        long expiresIn
) {
    // Compact constructor to set default values
    public AuthenticationResponse {
        if (tokenType == null) {
            tokenType = "Bearer";
        }
    }
}

File: auth-kotlin/src/main/kotlin/com/harrish/auth/dto/AuthenticationResponse.kt
package com.harrish.auth.dto

import com.fasterxml.jackson.annotation.JsonProperty

data class AuthenticationResponse(
    @JsonProperty("access_token")
    val accessToken: String,

    @JsonProperty("refresh_token")
    val refreshToken: String,

    @JsonProperty("token_type")
    val tokenType: String = TOKEN_TYPE,

    @JsonProperty("expires_in")
    val expiresIn: Long
) {
    companion object {
        private const val TOKEN_TYPE = "Bearer"
    }
}
